python/cudf/cudf/core/dataframe.py

```diff
@@ -2627,7 +2628,46 @@ def reindex(
 
         return DataFrame(cols, idx)
 
-    def set_index(self, index, drop=True):
+    def _set_index(
+        self, index, to_drop=None, inplace=False, verify_integrity=False,
+    ):
+        """Helper for `.set_index`
+
+        Parameters
+        ----------
+        index : Index
+            The new index to set.
+        to_drop : list
+            The list of labels indicating columns to drop
+        inplace : boolean
+            Modify the DataFrame in place (do not create a new object)
+        verify_integrity : boolean
+            Check for duplicates in the new index
+        """
+        if not isinstance(index, Index):
+            raise ValueError("Parameter index should be type `Index`.")
+
+        df = self if inplace else self.copy(deep=True)
+
+        if verify_integrity and not index.is_unique:
+            # TODO: indicate duplicate keys
+            raise ValueError("Index is not unique.\n {}".format(index))
```

Comment:
```suggestion
            raise ValueError(f"Index is not unique: {index}")
```

Stale TODO?

Reply:
Actually I want to indicate exactly which keys are duplicates here. Not just printing out the entire index. I haven't figured out how to do that yet. Do you have some suggestions?

Reply:
> Actually I want to indicate exactly which indices are duplicates here.

I don't think that is necessary here, we just print a generic message that Index contains non-unique values else-where in the code-base in similar instances like this.

https://github.com/rapidsai/cudf/blob/branch-0.16/python/cudf/cudf/core/dataframe.py#L342
https://github.com/rapidsai/cudf/blob/branch-0.16/python/cudf/cudf/core/reshape.py#L34